app/client/components/filterModule/FilterModule.jsx

```diff
@@ -124,15 +138,30 @@ class FilterModule extends React.Component {
           style={{ width: 200 }}
           onChange={this.updateGroupBy}
         >
-          {allowedGroups
-            ? Object.keys(allowedGroups).map(key => {
+          if(ungroupedGroups)
+          {ungroupedGroups.map(group => {
+            return <Option key={group}>{allowedGroups[group]["name"]}</Option>;
+          })}
+          if(senderGroups)
+          {
+            <OptGroup label={"Sender"}>
+              {senderGroups.map(group => {
+                return (
+                  <Option key={group}>{allowedGroups[group]["name"]}</Option>
+                );
+              })}
+            </OptGroup>
+          }
+          if(recipientGroups)
+          {
+            <OptGroup label={"Recipient"}>
+              {recipientGroups.map(group => {
                 return (
-                  <Option key={key}>
-                    {toTitleCase(replaceUnderscores(key))}
-                  </Option>
+                  <Option key={group}>{allowedGroups[group]["name"]}</Option>
                 );
-              })
-            : null}
+              })}
+            </OptGroup>
```

Comment:
Taking a hint out of @enjeyw OptGroup PR, can we add showing the group name once selected. See example from `filter.jsx`

```
let subList = keys.map(key => {
      //Here we show the label without the group in the dropdown, but with the group once selected
      let label = replaceUnderscores(possibleFilters[key]["name"] || key);
      return (
        <Option key={key} label={label}>
          {label.replace(userGroup, "")}
        </Option>
      );
    });
```

Reply:
Fixed!